cmd/tnf/addclaim/addclaim.go

```diff
@@ -27,12 +29,257 @@ var (
+				continue
+			} // if they are the same name
+			if result2.Status != result1.Status {
+				diffResult = append(diffResult, result1)
+			}
+		}
+	}
+	notFoundtestIn1, notFoundtestIn2 = missing(testcaseR1, testcaseR2)
+	return diffResult, removeDuplicateValues(notFoundtestIn1), removeDuplicateValues(notFoundtestIn2)
+}
+
+// empty struct (0 bytes)
+type void struct{}
+
+// missing compares two slices and returns slice of differences, between 2 sides
+func missing(a, b []string) (diffsAfromB, diffsBfromA []string) {
+	// create map with length of the 'a' slice
+	ma := make(map[string]void, len(a))
+	mb := make(map[string]void, len(b))
+	// Convert first slice to map with empty struct (0 bytes)
+	for _, ka := range a {
+		ma[ka] = void{}
+	}
+	// Convert first slice to map with empty struct (0 bytes)
+	for _, ka := range b {
+		mb[ka] = void{}
+	}
+	// find missing values in b
+	for _, kb := range b {
+		if _, ok := ma[kb]; !ok {
+			diffsAfromB = append(diffsAfromB, kb)
+		}
+	}
+	for _, ka := range a {
+		if _, ok := mb[ka]; !ok {
+			diffsBfromA = append(diffsBfromA, ka)
+		}
+	}
+	return diffsAfromB, diffsBfromA
+}
+
+func removeDuplicateValues(intSlice []string) []string {
+	keys := make(map[string]bool)
+	list := []string{}
+	// If the key(values of the slice) is not equal
+	// to the already present value in new slice (list)
+	// then we append it. else we jump on another element.
+	for _, entry := range intSlice {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			list = append(list, entry)
+		}
+	}
+	return list
+}
+
+// compare between 2 cni objects and print the diffrents
+func compare2cni(cni1, cni2 map[string][]Cni) {
+	for node, val := range cni1 {
+		for node2, val2 := range cni2 {
+			if node != node2 {
+				continue
+			}
+			diffPlugIn, notFoundNamesIn1, notFoundNamesIn2 := compare2cniHelper(val, val2, node2)
+			if len(notFoundNamesIn1) != 0 && notFoundNamesIn1 != nil {
+				log.Info("in node ", node2, " cnis found in claim1 but not present in claim2: ", notFoundNamesIn1)
+			}
+			if len(notFoundNamesIn2) != 0 && notFoundNamesIn2 != nil {
+				log.Info("in node ", node2, " cnis found in claim2 but not present in claim1: ", notFoundNamesIn2)
+			}
+			if len(diffPlugIn) != 0 {
+				log.Info("in node ", node2, " cnis present in both claim 1 and 2 but with different plugins: ", diffPlugIn)
+			}
+			break
+		}
+	}
+}
+
+// receiving 2 cnistruct and return :
+// 1. name of cni's that have same name but the plugin value are different - diffPlugins
+// 2. name of cni's that found on claim2 but not in claim1 - notFoundNamesIn1
+// 3. name of cni's that found on claim1 but not in claim2 - notFoundNamesIn3
+func compare2cniHelper(cniList1, cniList2 []Cni, node string) (diffPlugins []Cni, notFoundNamesIn1, notFoundNamesIn2 []string) {
+	var cniList1Name, cniList2Name []string
+	if len(cniList1) == 0 {
+		log.Info("in node ", node, " cnis present in claim2 and on claim1 that node dont have cni values", cniList2)
```

Comment:
```suggestion
		log.Infof("in node %s cnis present in claim2 and on claim1 that node dont have cni values: %v", node, cniList2)
```

There are a few other sections like this.